clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -75,17 +75,13 @@
     INVALID_REQUIRED_ACKS(21,
             new InvalidRequiredAcksException("Produce request specified an invalid value for required acks.")),
     ILLEGAL_GENERATION(22,
-            new IllegalGenerationException("Specified consumer generation id is not valid.")),
-    INCONSISTENT_PARTITION_ASSIGNMENT_STRATEGY(23,
-            new ApiException("The request partition assignment strategy does not match that of the group.")),
-    UNKNOWN_PARTITION_ASSIGNMENT_STRATEGY(24,
-            new ApiException("The request partition assignment strategy is unknown to the broker.")),
-    UNKNOWN_CONSUMER_ID(25,
-            new UnknownConsumerIdException("The coordinator is not aware of this consumer.")),
+            new IllegalGenerationException("Specified group generation id is not valid.")),
+    INCONSISTENT_GROUP_PROTOCOL(23,
+            new ApiException("The group member's supported protocols are incompatible with those of existing members.")),
+    UNKNOWN_MEMBER_ID(25,
+            new UnknownMemberIdException("The coordinator is not aware of this member.")),
     INVALID_SESSION_TIMEOUT(26,
             new ApiException("The session timeout is not within an acceptable range.")),
-    COMMITTING_PARTITIONS_NOT_ASSIGNED(27,
```

Comment:
If we are not assigning a different error code here, maybe we want to shift the rest error codes left? And same above for 23 / 24 / 25.